tests/cstor/invalidconfig/invalidconfig_test.go

```diff
@@ -0,0 +1,231 @@
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package volume
+
+import (
+	"strconv"
+	"strings"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	apis "github.com/openebs/maya/pkg/apis/openebs.io/v1alpha1"
+	pvc "github.com/openebs/maya/pkg/kubernetes/persistentvolumeclaim/v1alpha1"
+	sc "github.com/openebs/maya/pkg/kubernetes/storageclass/v1alpha1"
+	spc "github.com/openebs/maya/pkg/storagepoolclaim/v1alpha1"
+	"github.com/openebs/maya/tests/cstor"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+var _ = Describe("[cstor] [-ve] TEST INVALID STORAGEPOOLCLAIM", func() {
+	var (
+		err error
+	)
+
+	AfterEach(func() {
+		By("deleting resources created for testing cstor volume provisioning", func() {
+
+			By("listing spc")
+			spcList, err = ops.SPCClient.List(metav1.ListOptions{})
+			Expect(err).To(BeNil(), "while listing spc clients", spcList)
+
+			By("deleting spc")
+			for _, spc := range spcList.Items {
+				_, err = ops.SPCClient.Delete(spc.Name, &metav1.DeleteOptions{})
+				Expect(err).To(BeNil(), "while deleting the spc's", spc)
+			}
+			time.Sleep(5 * time.Second)
+		})
+	})
+
+	When("creating storagepoolclaim with invalid disk type", func() {
+		It("should not create any cstorpool", func() {
+
+			By("building spc object")
+			spcObj = spc.NewBuilder().
+				WithName(spcName).
+				WithDiskType("invalid-disk-type").
+				WithMaxPool(cstor.PoolCount).
+				WithOverProvisioning(false).
+				WithPoolType(string(apis.PoolTypeStripedCPV)).
+				Build().Object
+
+			By("creating storagepoolclaim")
+			_, err = ops.SPCClient.Create(spcObj)
+			Expect(err).To(BeNil(), "while creating spc", spcName)
+
+			By("verifying healthy csp count as 0")
+			cspCount := ops.GetHealthyCSPCount(spcName, cstor.PoolCount)
+			Expect(cspCount).To(Equal(0), "while checking cstorpool health status")
+
+		})
+	})
+
+	When("creating storagepoolclaim with invalid pool type", func() {
+		It("should not create any cstorpool", func() {
+
+			By("building spc object")
+			spcObj = spc.NewBuilder().
+				WithName(spcName).
+				WithDiskType(string(apis.TypeSparseCPV)).
+				WithMaxPool(cstor.PoolCount).
+				WithOverProvisioning(false).
+				WithPoolType(string("invalid-pool-type")).
+				Build().Object
+
+			By("creating storagepoolclaim")
+			_, err = ops.SPCClient.Create(spcObj)
+			Expect(err).To(BeNil(), "while creating spc", spcName)
+
+			By("verifying healthy csp count as 0")
+			cspCount := ops.GetHealthyCSPCount(spcName, cstor.PoolCount)
+			Expect(cspCount).To(Equal(0), "while checking cstorpool health status")
+
+		})
+	})
+
+	When("creating storagepoolclaim with invalid pool count", func() {
+		It("should not create any cstorpool", func() {
+
+			By("building spc object")
```

Comment:
same comment.
Make use of storagepoolclaim instead of spc.
make use of expanded names when the entire message is not long